src/controllers/campaign/delete.ts

```diff
@@ -0,0 +1,27 @@
+import { AppResponse } from '@/common/utils';
+import AppError from '@/common/utils/appError';
+import { catchAsync } from '@/middlewares';
+import { Request, Response } from 'express';
+import { campaignModel as campaign } from '@/models/campaignModel';
+import { DateTime } from 'luxon';
+
+const deleteCampaign = catchAsync(async (req: Request, res: Response) => {
+	const { user } = req;
+
+	if (!user) {
+		throw new AppError('User not found!. Please login');
+	}
+
+	const { campaignId } = req.params;
+	const updatedCampaign = await campaign.findByIdAndUpdate(campaignId, {
```

Comment:
this implementation allows for any authenticated user to delete another person's campaign provided they know the campaign id, add a check to ensure the currently logged in user ID matches the campaign creator user ID